src/commands/createHttpFunction.ts

```diff
@@ -61,7 +61,18 @@ function generateSuffixedName(preferredName: string, i: number): string {
 }
 
 async function isNameAvailable(folderPath: string, newName: string): Promise<boolean> {
-    return !(await fse.pathExists(path.join(folderPath, newName)));
+    // if the file or folder doesn't exist, it will throw an error so we have to wrap these in try/catches
+    try {
+        await fse.access(path.join(folderPath, newName));
+    } catch (error) {
+        try {
+            return !(await fse.readdir(folderPath)).find(file => { return file.includes(`${newName}.`); });
```

Comment:
I think this has several edge cases it doesn't handle well. For example, maybe I have a file named "HttpTrigger.test.cs" and I want to add "HttpTrigger.cs". Remind me again why you have to prompt for name in this extension instead of letting the Functions extension do it? It's just so much easier to validate the name once the Functions extension knows which language it is

Reply:
I think it was because we wanted to have the generic name "endpoint1", but honestly, I think it makes a lot more sense to just let Functions handle it.